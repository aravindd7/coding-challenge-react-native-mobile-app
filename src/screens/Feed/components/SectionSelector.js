import React from "react";
import PropTypes from "prop-types";

import { Text, View, TouchableOpacity, StyleSheet } from "react-native";

class SectionSelector extends React.PureComponent {
  _changeSection_New = () => this.props.feedSectionChange("new");
  _changeSection_Top = () => this.props.feedSectionChange("top");
  _changeSection_Controversial = () =>
    this.props.feedSectionChange("controversial");
  _changeSection_Hot = () => this.props.feedSectionChange("hot");

  render() {
    const { section } = this.props;
    return (
      <View style={styles.container}>
        <TouchableOpacity
          onPress={this._changeSection_New}
          style={[styles.item, section == "new" ? styles.selected : {}]}
        >
          <Text
            style={[styles.text, section == "new" ? styles.selected_text : {}]}
          >
            New
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={this._changeSection_Top}
          style={[styles.item, section == "top" ? styles.selected : {}]}
        >
          <Text
            style={[styles.text, section == "top" ? styles.selected_text : {}]}
          >
            Top
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={this._changeSection_Controversial}
          style={[
            styles.item,
            section == "controversial" ? styles.selected : {}
          ]}
        >
          <Text
            style={[
              styles.text,
              section == "controversial" ? styles.selected_text : {}
            ]}
          >
            Popular
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={this._changeSection_Hot}
          style={[styles.item, section == "hot" ? styles.selected : {}]}
        >
          <Text
            style={[styles.text, section == "hot" ? styles.selected_text : {}]}
          >
            Hot
          </Text>
        </TouchableOpacity>
      </View>
    );
  }
}
SectionSelector.propTypes = {
  section: PropTypes.string.isRequired,
  feedSectionChange: PropTypes.func.isRequired
};

export default SectionSelector;
const styles = StyleSheet.create({
  container: {
    backgroundColor: "#FFFFFF",
    flexDirection: "row",
    alignItems: "stretch",
    justifyContent: "space-between",
    borderBottomWidth: 1,
    borderBottomColor: "#CCCCCC"
  },
  item: {
    padding: 10,
    flex: 1
  },
  text: {
    fontSize: 12,
    textAlign: "center",
    color: "#47b8ff"
  },
  selected: {
    backgroundColor: "#47b8ff"
  },
  selected_text: {
    color: "#FFFFFF"
  }
});
