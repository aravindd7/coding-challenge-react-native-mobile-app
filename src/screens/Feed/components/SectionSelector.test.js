import React from "react";

import ShallowRenderer from "react-test-renderer/shallow"; // ES6

import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallow } from "enzyme";

import SectionSelector from "./SectionSelector";

configure({ adapter: new Adapter() });

const props = {
  section: "top",
  feedSectionChange: () => {}
};

it("match on Shallow mode new", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<SectionSelector {...props} section="new" />);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});

it("match on Shallow mode top", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<SectionSelector {...props} section="top" />);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});

it("match on Shallow mode controversial", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<SectionSelector {...props} section="controversial" />);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});

it("match on Shallow mode hot", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<SectionSelector {...props} section="hot" />);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});

it("call methods", () => {
  const feedSectionChangeMock = jest.fn();

  const wrapper = shallow(
    <SectionSelector {...props} feedSectionChange={feedSectionChangeMock} />
  );
  wrapper.instance()._changeSection_New();
  wrapper.instance()._changeSection_Top();
  wrapper.instance()._changeSection_Controversial();
  wrapper.instance()._changeSection_Hot();

  expect(feedSectionChangeMock.mock.calls).toMatchSnapshot();
});
