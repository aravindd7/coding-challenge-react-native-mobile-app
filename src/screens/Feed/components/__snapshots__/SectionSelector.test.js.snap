// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`call methods 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "new",
    ],
    Array [
      "top",
    ],
    Array [
      "controversial",
    ],
    Array [
      "hot",
    ],
  ],
}
`;

exports[`match on Shallow mode controversial 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#CCCCCC",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#47b8ff",
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      New
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#47b8ff",
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      Top
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
        },
        Object {
          "backgroundColor": "#47b8ff",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#47b8ff",
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      Popular
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#47b8ff",
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      Hot
    </Text>
  </TouchableOpacity>
</View>
`;

exports[`match on Shallow mode hot 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#CCCCCC",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#47b8ff",
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      New
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#47b8ff",
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      Top
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#47b8ff",
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      Popular
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
        },
        Object {
          "backgroundColor": "#47b8ff",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#47b8ff",
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      Hot
    </Text>
  </TouchableOpacity>
</View>
`;

exports[`match on Shallow mode new 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#CCCCCC",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
        },
        Object {
          "backgroundColor": "#47b8ff",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#47b8ff",
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      New
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#47b8ff",
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      Top
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#47b8ff",
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      Popular
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#47b8ff",
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      Hot
    </Text>
  </TouchableOpacity>
</View>
`;

exports[`match on Shallow mode top 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#CCCCCC",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#47b8ff",
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      New
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
        },
        Object {
          "backgroundColor": "#47b8ff",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#47b8ff",
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      Top
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#47b8ff",
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      Popular
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
        },
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#47b8ff",
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      Hot
    </Text>
  </TouchableOpacity>
</View>
`;
